c_models/reinjector/reinjector.c

```diff
@@ -230,6 +240,29 @@ INT_HANDLER dropped_packet_callback() {
         // check for overflow from router
         if (rtr_dstat & RTR_DOVRFLW_MASK) {
             n_missed_dropped_packets += 1;
+        } else {
+
+            // Note that the processor_dump and link_dump flags are sticky
+            // so you can only really count these if you *haven't* missed a
+            // dropped packet - hence this being split out
+
+            if (is_processor_dump > 0) {
+
+                // add to the count the number of active bits from this dumped
+                //packet, as this indicates how many processors this packet
+                // was meant to go to.
+                n_processor_dumped_packets +=
+                    __builtin_popcount(is_processor_dump);
+            }
+
+            if (is_link_dump > 0) {
+
+                // add to the count the number of active bits from this dumped
+                //packet, as this indicates how many links this packet was
+                // meant to go to.
+                n_link_dumped_packets +=
+                    __builtin_popcount(is_link_dump);
+            }
         }
 
         // Only update this counter if this is a packet to reinject
```

Comment:
I have moved this out of the "else" as I think you can still capture a dropped packet even if the overflow is marked.  If you know that this isn't the case, please shout!